vsintegration/src/vs/FsPkgs/FSharp.Project/FS/AppConfigHelper.fs

```diff
@@ -317,92 +317,94 @@ namespace Microsoft.VisualStudio.FSharp.ProjectSystem
             if dirty then isDirty <- dirty            
         
         /// Updates list of binding redirects to the config file. Content of list is governed by the major version of target framework.
-        /// Current redirects: 2.0 -> "2.0.0.0" to "2.3.0.0". 4.0 or greater - "2.0.0.0" to "4.3.0.0", "2.3.5.0" to "4.3.0.0", "4.0.0.0" to "4.3.0.0"
-        member x.EnsureHasBindingRedirects(majorVersion) =
+        member x.EnsureHasBindingRedirects(majorVersion, minorVersion, targetFSharpCoreVersion, autoGenerateBindingRedirects) =
 
-            // some helpers to simplify work with XLinq
-            let xname = System.Xml.Linq.XName.Get
-            let xnameAsmV1 name = xname ("{urn:schemas-microsoft-com:asm.v1}" + name)
+            if not autoGenerateBindingRedirects then
+                // some helpers to simplify work with XLinq
+                let xname = System.Xml.Linq.XName.Get
+                let xnameAsmV1 name = xname ("{urn:schemas-microsoft-com:asm.v1}" + name)
 
-            let fsCoreAttributes = 
-                [
-                    xname "name", "FSharp.Core"
-                    xname "publicKeyToken", Microsoft.VisualStudio.FSharp.ProjectSystem.Utilities.FsCorePublicKeyToken
-                    xname "culture", "neutral"
-                ]
-
-            // depending on major version of target framework we need to populate corresponding binding redirects in config
-            let redirects = 
-                if majorVersion >= 4 then
-                    [
-                        "2.0.0.0", "4.3.0.0"
-                        "2.3.5.0", "4.3.0.0"
-                        "4.0.0.0", "4.3.0.0"
-                    ]
-                else
+                let fsCoreAttributes = 
                     [
-                        "2.0.0.0", "2.3.0.0"
+                        xname "name", "FSharp.Core"
+                        xname "publicKeyToken", Microsoft.VisualStudio.FSharp.ProjectSystem.Utilities.FsCorePublicKeyToken
+                        xname "culture", "neutral"
                     ]
 
-            let OldVersion = "oldVersion"
-            let NewVersion = "newVersion"
-            let BindingRedirect = "bindingRedirect"
-            let DependentAssembly = "dependentAssembly"
-            let AssemblyIdentity = "assemblyIdentity"
+                // depending on major version of target framework we need to populate corresponding binding redirects in config
+                let redirects = 
+                    if majorVersion >= 4 then
+                        [
+                            "2.0.0.0", targetFSharpCoreVersion
```

Comment:
The existing code didn't redirect, but I agree there is a good argument for redirecting the portables that also target the desktop ... which is to say all of them.

Good eyes.